src/java.base/share/classes/java/lang/Module.java

```diff
@@ -320,13 +319,16 @@ void ensureNativeAccess(Class<?> owner, String methodName) {
                 // warn and set flag, so that only one warning is reported per module
                 String cls = owner.getName();
                 String mtd = cls + "::" + methodName;
-                String mod = isNamed() ? "module " + getName() : "the unnamed module";
+                String mod = isNamed() ? "module " + getName() : "an unnamed module";
                 String modflag = isNamed() ? getName() : "ALL-UNNAMED";
+                String caller = currentClass != null ?
+                        " by " + currentClass.getName() : "";
                 System.err.printf("""
                         WARNING: A restricted method in %s has been called
-                        WARNING: %s has been called by %s
-                        WARNING: Use --enable-native-access=%s to avoid a warning for this module
-                        %n""", cls, mtd, mod, modflag);
+                        WARNING: %s has been called%s in %s
```

Comment:
```suggestion
                        WARNING: %s has been called by %s in %s
```
?

Reply:
The current code does the right thing, since in some cases the caller is `null` and the second `%s` should expand to an empty string. So  in the `caller == null` case, the message becomes:

    Class::method has been called in an unnamed module

There was also an offline suggestion to change it to:

    Class::method has been called by code in an unnamed module
    
Which I think is a good idea.